File: Data Structures & Algorithms/generate-parentheses/submission-1.cpp
class Solution {
public:
    void branch(int open, int close, int n, string& stk, vector<string>& res){
        if(open == n && close == open){
            res.push_back(stk);
            return;
        }

        if(open > close){
            stk += ")";
            branch(open, close + 1, n, stk, res);
            stk.pop_back();
        }
        if(open < n){
            stk += "(";
            branch(open + 1, close, n, stk, res);
            stk.pop_back();
        }
    }
    vector<string> generateParenthesis(int n) {
        string stk;
        vector<string> res;
        branch(0, 0, n, stk, res);
        return res;
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-1.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        res = 0

        for i in range(n):
            p1, p2 = i, i
            while 0 <= p1 < n and 0 <= p2 < n and s[p1] == s[p2]:
                res += 1
                p1 -= 1
                p2 += 1
            
            p1, p2 = i, i + 1
            while 0 <= p1 < n and 0 <= p2 < n and s[p1] == s[p2]:
                res += 1
                p1 -= 1
                p2 += 1
        return res
        
